### ISSUE TITLE
Make usage of the `search` icon more consistent and meaningful

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63023]
### Description

Similarly to https://github.com/WordPress/gutenberg/issues/62873, to provide cohesive design language in the editor UI the _usage_ of icons should be consistent. Icons should identify unique features or, at least, similar features.

The `search` icon is one of the most popular and familiar icons on the web. I'm not sure it's ideal to use it for features other than, well, a search.

In the Site editor it is used also for the Command palette. This is potentially confusing for users and particularly disorienting when the UI shows _anoter search icon_ close to the command palette one. Actually, in some cases, the editor shows two very close search icons that are associated with entirely different features. Example screenshot:

![Screenshot 2024-07-01 at 12 31 48](https://github.com/WordPress/gutenberg/assets/1682452/6aa5609d-713d-4aea-9a2a-7280d11c7ebe)

I'd tend to think the command palette should not use the search icon in the first place.

Another occurrence of `icon={ search }` is in the Image editor. When editing an image, the block toolbar shows a `search` icon that is associated with the `Zoom` feature. Screenshot:

![Screenshot 2024-07-01 at 14 36 06](https://github.com/WordPress/gutenberg/assets/1682452/bb8b57ff-06bb-43c0-b427-4b30cb2532f3)

Also in this case I'd argue the search icon isn't the best one to use. Usually, zoom icons use a lens that either contains either a plus or minus sign. Example from the Material design icons:

![Screenshot 2024-07-01 at 14 41 52](https://github.com/WordPress/gutenberg/assets/1682452/445adbda-ccb8-490d-8e40-92785226e4f8)

### Step-by-step reproduction instructions

- Go to the Site editor > Pages.
- Observe that, regardless of whether the active view is List, Table, or Grid, the editor UI shows two search icon for two different features.
- Add an Image block, set an image, and click `Crop` in the block toolbar.
- Observe the toolbar shows a `Zoom` control with a search icon.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
